frontend/src/metabase/core/components/FileInput/FileInput.tsx

```diff
@@ -0,0 +1,66 @@
+import React, {
+  ChangeEvent,
+  FocusEvent,
+  forwardRef,
+  HTMLAttributes,
+  Ref,
+  useCallback,
+  useState,
+} from "react";
+import { t } from "ttag";
+import { InputButton, InputField, InputRoot } from "./FileInput.styled";
+
+export type FileInputAttributes = Omit<
+  HTMLAttributes<HTMLLabelElement>,
+  "onChange" | "onFocus" | "onBlur"
+>;
+
+export interface FileInputProps extends FileInputAttributes {
+  className?: string;
+  name?: string;
+  autoFocus?: boolean;
+  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
+  onFocus?: (event: FocusEvent<HTMLInputElement>) => void;
+  onBlur?: (event: FocusEvent<HTMLInputElement>) => void;
+}
+
+const FileInput = forwardRef(function FileInput(
+  {
+    className,
+    name,
+    autoFocus,
+    onChange,
+    onFocus,
+    onBlur,
+    ...props
+  }: FileInputProps,
+  ref: Ref<HTMLLabelElement>,
+): JSX.Element {
+  const [hasValue, setHasValue] = useState(false);
+
+  const handleChange = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      const { files } = event.target;
+      setHasValue(files != null && files.length > 0);
+      onChange && onChange(event);
+    },
+    [onChange],
+  );
+
+  return (
+    <InputRoot innerRef={ref as any} className={className} {...props}>
```

Comment:
nip: any reason to pick `className` from `props`? Seems like it's not used anywhere and it could be passed with `props` spreading